seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/OrcReadStrategy.java

```diff
@@ -80,6 +86,225 @@
+                    reader.getStripes().size(),
+                    reader.getNumberOfRows());
+            long rowCountPerSplit = rowCountPerSplitByUser;
+            if (rowCountPerSplit <= 0) {
+                // auto split by file size
+                long fileSize = reader.getContentLength();
+                long rowCount = reader.getNumberOfRows();
+                long splitCount =
+                        fileSize
+                                / (fileSizePerSplitByUser <= 0
+                                        ? DEFAULT_FILE_SIZE_PER_SPLIT
+                                        : fileSizePerSplitByUser);
+                if (rowCount == 0 || splitCount == 0) {
+                    log.warn("cannot get file size or row count for orc file:{}", path);
+                    rowCountPerSplit = DEFAULT_ROW_COUNT;
+                } else {
+                    rowCountPerSplit = rowCount / splitCount;
+                }
+            }
+            long low = 0;
+            long high = low;
+            int splitCountAll = 0;
+            if (reader.getStripes() == null) {
+                log.warn("cannot get stripes for orc file:{}", path);
+                fileSourceSplits.add(new FileSourceSplit(path));
+                return fileSourceSplits;
+            }
+            for (int i = 0; i < reader.getStripes().size(); i++) {
+                StripeInformation stripe = reader.getStripes().get(i);
+                long leftOverCount = stripe.getNumberOfRows();
+                int splitCount4Strip = 0;
+                long startRow = low;
+                while (leftOverCount > 0) {
+                    if (leftOverCount > rowCountPerSplit) {
+                        high = low + rowCountPerSplit;
+                    } else {
+                        high = low + leftOverCount;
+                    }
+                    FileSourceSplit split = new FileSourceSplit(path, low, high - 1);
+                    fileSourceSplits.add(split);
+                    leftOverCount = leftOverCount - rowCountPerSplit;
+                    low = high;
+                    splitCountAll++;
+                    splitCount4Strip++;
+                }
+                log.debug(
+                        "generate split count:{} for this strip:{}, startRow:{}, endRow:{},",
+                        splitCount4Strip,
+                        i,
+                        startRow,
+                        high - 1);
+            }
+            log.info("generate split count:{} for this file:{}", splitCountAll, path);
+            return fileSourceSplits;
+        } catch (IOException e) {
+            String errorMsg = String.format("Create orc reader for this file [%s] failed", path);
+            throw new FileConnectorException(
+                    CommonErrorCodeDeprecated.READER_OPERATION_FAILED, errorMsg);
+        }
+    }
+
+    @Override
+    public void read(FileSourceSplit split, Collector<SeaTunnelRow> output)
```

Comment:
It would be best to do some refactoring, I found that the new read method has a lot in common with the old read method.